iXBRLViewerPlugin/viewer/src/js/reportset.js

```diff
@@ -0,0 +1,230 @@
+            for (const reportData of sourceReport.targetReports) {
+                const report = new XBRLReport(this, reportData);
+                this.reports.push(report);
+                for (const [id, factData] of Object.entries(reportData.facts)) {
+                    const vuid = viewerUniqueId(reportIndex, id);
+                    this._items[vuid] = new Fact(report, vuid, factData);
+                }
+
+                // Now resolve footnote references, creating footnote objects for "normal"
+                // footnotes, and finding Fact objects for fact->fact footnotes.  
+                //
+                // Associate source facts with target footnote/facts to allow two way
+                // navigation.
+                for (const [id, factData] of Object.entries(reportData.facts)) {
+                    const vuid = viewerUniqueId(reportIndex, id);
+                    const fact = this._items[vuid];
+                    const fns = factData.fn || [];
+                    fns.forEach((fnid) => {
+                        const fnvuid = viewerUniqueId(reportIndex, fnid);
+                        var fn = this._items[fnvuid];
+                        if (fn === undefined) {
+                            fn = new Footnote(fact.report, fnvuid, "Footnote " + (fnorder.indexOf(fnvuid) + 1));
+                            this._items[fnvuid] = fn;
+                        }
+                        // Associate fact with footnote
+                        fn.addLinkedFact(fact);
+                        fact.addFootnote(fn);
+                    });
+                }
+            }
+        }
+    }
+
+    availableLanguages() {
+        return Array.from(this.reports.reduce(
+            (langs, report) => new Set([...langs, ...report.availableLanguages()]), 
+            new Set()
+        ));
+    }
+
+    getItemById(id) {
+        return this._items[id];
+    }
+
+    getIXNodeForItemId(id) {
+        return this._ixNodeMap[id] || {};
+    }
+
+    facts() {
+        return Object.values(this._items).filter(i => i instanceof Fact);
+    }
+
+    filingDocuments() {
+        return this._data.filingDocuments;
+    }
+
+    prefixMap() {
+        return this._data.prefixes;
+    }
+
+    namespaceGroups() {
+        const counts = {};
+        $.each(this.facts(), function (i, f) {
+            counts[f.conceptQName().prefix] = counts[f.conceptQName().prefix] || 0;
+            counts[f.conceptQName().prefix]++;
+        });
+        const prefixes = Object.keys(counts);
+        prefixes.sort(function (a, b) { return counts[b] - counts[a] });
+        return prefixes;
+    }
+
+    getUsedPrefixes() {
+        if (this._usedPrefixes === undefined) {
+            this._usedPrefixes = new Set(Object.values(this._items)
+                    .filter(f => f instanceof Fact)
+                    .map(f => f.getConceptPrefix()));
```

Comment:
Pre-existing, I guess this method is really `getUsedFactPrefixes` since iit doesn't reflect prefixes used by dimensions, members, outline, etc.

Reply:
Have renamed to `getUsedConceptPrefixes`